provider/provider.go

```diff
@@ -21,9 +21,51 @@ func init() {
 
 func New(version string) *schema.Provider {
 	p := &schema.Provider{
-
 		Schema: map[string]*schema.Schema{
 			esKeyName: providerSchema.GetConnectionSchema(esKeyName, true),
+			"kibana": {
+				Description: "Kibana connection configuration block.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"username": {
+							Description:  "Username to use for API authentication to Kibana.",
+							Type:         schema.TypeString,
+							Optional:     true,
+							DefaultFunc:  schema.EnvDefaultFunc("ELASTICSEARCH_USERNAME", nil),
+							RequiredWith: []string{"kibana.0.password"},
+							ExactlyOneOf: []string{"elasticsearch.0.username", "kibana.0.username"},
```

Comment:
This doesn't seem right. We should default to using the ES credentials if no Kibana credentials are supplied, but people should be able to use one account with ES and another account for Kibana.

Reply:
This was my initial attempt at finding a way to support [example 1 and 2 here described in the description](https://github.com/elastic/terraform-provider-elasticstack/issues/100), but I wasn't sure so just left it to move forward on integrating the go API. for Kibana.

I'm happy to jump on a call to take stock of where things are before moving forward though to make sure we're on the same page?